roll/testing.py

```diff
@@ -36,12 +45,15 @@ def encode_body(self, body, headers):
             headers['Content-Type'] = content_type
         body, headers = self.encode_body(body, headers)
         protocol = self.app.factory()
+        protocol.connection_made(Transport())
         protocol.on_message_begin()
         protocol.on_url(path.encode())
         protocol.request.body = body
         protocol.request.method = method
         protocol.request.headers = headers
-        return await self.app(protocol.request, protocol.response)
+        await self.app(protocol.request, protocol.response)
```

Comment:
I suggest to extract this change to a separate commit. IMHO this should be merged right away.

Reply:
Is that this urgent? #lazyme